packages/remix-react/scroll-restoration.tsx

```diff
@@ -47,6 +49,12 @@ export function ScrollRestoration({
     []
   );
 
+  // In SPA Mode, there's nothing to restore on initial render since we didn't
+  // render anything on the server
+  if (isSpaMode) {
+    return null;
```

Comment:
We don't need to do initial SSR scroll restoration in SPA mode since no content was SSR'd